orion/sites/admin/src/location/location.js

```diff
@@ -20,29 +20,79 @@ export class LocationPage extends React.Component {
         });
     }
 
+    onClickSelectAll() {
+        var items = document.getElementsByName("unpublished");
+        for (var i = 0; i < items.length; i++) {
+            if (items[i].type == "checkbox") {
+                items[i].checked = true;
+            }
+        }
+    }
+
+    onClickPublish() {
+        console.log("clicked publish");
+    }
+
     render() {
         const location = this.state.list.map((location, index) => {
             return <LocationRow key={index} location={location} />;
         });
         const numLocations = location.length;
+        let numUnpublished = 0;
+        let numSelected = 0;
```

Comment:
I'm a bit stumped because the checkbox is located in a different class, and I'm not sure how I would change my variable in the LocationPage from a button being clicked in the LocationRow class. I tried making an instance of the classes to try and call functions between the two, but it's not working out :(

Reply:
Oh I see. Ok so if I'm understanding your problem at the moment, the checkbox is in `LocationRow` while the data (numSelected) is in `LocationPage`.
Remember how we did modals? We passed in a function to a Modal to be used when something happens (this is called callbacks if you remember). You can do something similar here. Define a function `onCheckRow`. It's defined in LocationPage. When you create LocationRow, pass in that function so it will look like this:

```
<LocationRow key={index} location={location} onCheckRow={this.onCheckRow}/>;
```

Once you do that, this onCheckRow function is available inside the LocationRow component! You can access it by using `this.props.onCheckRow`. I would probably an onClick of the checkbox to this new function. So it would look something like this:
```
<input type="checkbox" onClick={this.props.onCheckRow}>
```

So what I would do first is set this up so that `onCheckRow` is super simple like just console.log something.
When I click on a checkbox, we should see that print statement in the console.